jsx/Modal.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * This file contains the React Component for a Modal Window.
+ *
+ * @author Henri Rabalais
+ * @version 1.1.0
+ *
+ */
+import swal from 'sweetalert2';
+
+/**
+ * Modal Component.
+ * React wrapper for a Modal Window. Allows to dynamically toggle a Modal window.
+ *
+ * ================================================
+ * Usage:
+ * - Wrap the contents to be displayed by the Modal Window by the Modal Component.
+ * - Use the 'show' prop to toggle the Modal Component's presentation.
+ * - Use the 'title' prop to set a title for the Modal Component.
+ * - Use the 'onSubmit' prop set a submission function for the Modal's contents.
+ * - Use the 'throwWarning' prop to throw a warning upon closure of the Modal Window
+ * =================================================
+ *
+ */
+class Modal extends React.Component {
+  constructor() {
+    super();
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  closeModal() {
+    if (this.props.throwWarning) {
+      swal({
+        title: "Are You Sure?",
+        text: "Leaving the form will result in the loss of any information entered.",
+        type: "warning",
+        showCancelButton: true,
+        confirmButtonText: 'Proceed',
+        cancelButtonText: 'Cancel'
+      }).then(result => {
+        if (result.value) {
+          this.props.closeModal();
+        }
+      });
+    } else {
+      this.props.closeModal();
+    }
+  }
+
+  render() {
+    const headerStyle = {
+      display: 'flex',
```

Comment:
i agree with @davidblader . if all these stylings are constant, why not put them in the css file? if the variables are only providing styling, it may as well be made into a class that you would add to each elements className attribute no? then the css styling would be where they belong, the css file

Reply:
There's definitely an argument to be made in either direction (ref for list of styling methods: https://codeburst.io/4-four-ways-to-style-react-components-ac6f323da822) and I think at the end of the day it comes down to context and personal preference. I'm easy either way in this case, but I think this merits a discussion of LORIS-documented standards for react.

Reply:
i think we should definitely update and make more robust the LORIS react standards. in this case however, i think it really comes down to code organization. there's a reason why we have css files and css directories, and in this case, i think they should be utilized as intended. this is however, my own personal opinion

Reply:
Yes, code organisation definitely comes into question here. Precisely for that reason, an argument could be made that this styling code is well suited to stay within this class. The styling here is not made to be used by any other component; it is exclusive to the modal window. Keeping the css inline also provides context to the components. CSS styling sheets have a tendency of becoming a bit of an unorganized styling graveyard where it becomes difficult to pinpoint what styling is associated with what html. Keeping the code here makes it easy to track and update. 

On the other hand, massive changes to CSS across loris could benefit from having everything in a well organised style sheet. Having a bunch of css styling nested in different components could make the task of changing many things at once difficult.

I see benefits and disadvantages to both, but neither one is 'right' or 'wrong'. It would be nice to have a meeting to discuss certain standards (such as whether LORIS uses inline styling or not) so that there can be a precedent from which to review future code. We can then add this to the CodingStandards.md in https://github.com/aces/Loris/tree/master/docs, or create an entirely new file, for reference.